### PR TITLE
feat: per-version openapi artifact

### PR DESCRIPTION
**Pull request by epipav**
## Summary

- Adds a version registry (`api/src/versions.ts`) as the single source of truth for supported API versions (currently `v1`), used by both the app and the export script.
- Serves `GET /<version>/openapi.json` for every registered version. Each document is derived from the root swagger registration by filtering paths to that version's `/<version>/` prefix and setting `info.version` from the registry (`v1` → `1.0.0`, `v1-alpha` → `1.0.0-alpha`). The spec routes themselves are hidden from the documents.
- Reworks `scripts/export-openapi.ts` to write one artifact per registered version to `openapi/<version>.json`, fetching each document from the live route so exported files are byte-identical to what the server serves. The optional CLI argument now selects the output directory.
- `BuildAppOptions.versions` allows tests to override the registry; cross-version isolation is covered with `v1`/`v2` and `v1`/`v1-alpha` combinations.

When the per-version router trees land (IN-1137), adding a version becomes a one-line registry change.

## Test plan

- `cd api && pnpm test` — 56/56 green, including new `tests/openapi-versions.test.ts` and re-authored `tests/openapi-export.test.ts`
- `pnpm openapi:export` then compare `openapi/v1.json` with `curl localhost:4000/v1/openapi.json`

Jira: [IN-1138](https://linuxfoundation.atlassian.net/browse/IN-1138)

[IN-1138]: https://linuxfoundation.atlassian.net/browse/IN-1138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/scripts/export-openapi.ts [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 await app.ready();
-await mkdir(dirname(outPath), { recursive: true });
-await writeFile(outPath, JSON.stringify(app.swagger(), null, 2) + '\n');
+await mkdir(outDir, { recursive: true });
+
+for (const version of API_VERSIONS) {
+  // Fetch the served document (rather than app.swagger() directly) so the export
+  // is byte-for-byte identical to what /<version>/openapi.json actually serves.
+  const res = await app.inject({ method: 'GET', url: `/${version}/openapi.json` });
+  await writeFile(resolve(outDir, `${version}.json`), JSON.stringify(res.json(), null, 2) + '\n');
```
[the remark this answers is not in the document]
Comment (epipav) [reply api/scripts/export-openapi.ts]: Fixed in 132dc6f6: the script now fails on a non-200 response and writes `res.rawPayload` unchanged. Kept the test's parsed comparison, with raw bytes written parity holds by construction.

File: api/src/app.ts [hunk 2]
```diff
@@ -33,7 +36,28 @@ export async function buildApp(options: BuildAppOptions = {}): Promise<FastifyIn
...
-  app.get('/v1/openapi.json', async () => app.swagger());
+  const versionedDoc = (version: string) => {
+    const source = app.swagger();
+    // Exact prefix with trailing slash so e.g. /v1-alpha/ routes never leak into /v1's spec.
+    const pathPrefix = `/${version}/`;
+    return {
+      ...source,
+      info: { ...source.info, version: specVersionFor(version) },
+      paths: Object.fromEntries(
+        Object.entries(source.paths ?? {}).filter(([path]) => path.startsWith(pathPrefix)),
```
Comment (themarolt) [inline api/src/app.ts]: the isolation here only covers `paths` - `components` and any root-level `tags` come straight from the shared registration. no components today since nothing calls `addSchema`, but once we have shared models (or `/v1-alpha` living next to `/v1` per adr 0003) the `/v1` doc carries schemas that no `/v1` path references, and Scalar lists those under Models. do we want to prune components to the ones reachable from the filtered paths, or is that fine until it actually bites?

**Review by gaspergrom — APPROVED** [review]
The version registry and prefix filtering hold up under the cross-version test cases, including the v1/v1-alpha shared-prefix edge case. The export script now writes the raw response bytes, so the artifacts match what the server serves. Approving.

File: api/src/app.ts [hunk 3]
```diff
@@ -7,21 +7,81 @@ import fastifyStatic from '@fastify/static';
...
+  }
+  if (node && typeof node === 'object') {
+    for (const [key, value] of Object.entries(node)) {
+      if (key === '$ref' && typeof value === 'string') {
+        into.add(value);
+      } else {
+        collectRefs(value, into);
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (epipav) [reply api/src/app.ts:36]: fastify-swagger renames hoisted schemas to `def-N` but leaves `discriminator.mapping` strings untouched, so any mapping an author writes already dangles in the served doc before pruning even runs. Real support needs a `refResolver` that keeps `$id` component names, gonna add that with the first discriminated union we ship since nothing emits a discriminator today.
